bundles/org.openhab.core.io.monitor/src/main/java/org/openhab/core/io/monitor/internal/metrics/RuleMetric.java

```diff
@@ -80,9 +81,9 @@ public void unbind() {
         if (meterRegistry == null) {
             return;
         }
-        for (Meter meter : meterRegistry.getMeters()) {
+        for (Meter meter : Objects.requireNonNull(meterRegistry).getMeters()) {
```

Comment:
That's rather ugly. Can't you simply do a proper `else` case for the if-statement above and assign `meterRegistry` to a local variable to make the null checks happy?

Reply:
Unfortunately there is a clash with the io.micrometer.core.instrument.@NonNull annotation, which is not applicable to local variables. And this the least verbose variant of the whole class I got to compile without warnings.

Reply:
No, that's not the problem. What I meant is https://github.com/pravussum/openhab-core/pull/1.

Reply:
No, this is not the problem. What I meant is https://github.com/pravussum/openhab-core/pull/1.

Reply:
Hm, I think I tried exactly that and got some compiler error/warning that forced me to change the Nullable annotations (that's why I mentioned them), but I don't get it anymore. Never mind, probably an environment issue - your changes are merged, thanks for the taking the effort!